pandas/core/window/numba_.py

```diff
@@ -0,0 +1,110 @@
+import types
+from typing import Callable, Dict, Optional, Tuple
+
+import numpy as np
+
+from pandas.compat._optional import import_optional_dependency
+
+
+def make_rolling_apply(func, args, nogil, parallel, nopython):
+    numba = import_optional_dependency("numba")
+
+    if parallel:
+        loop_range = numba.prange
+    else:
+        loop_range = range
+
+    if isinstance(func, numba.targets.registry.CPUDispatcher):
+        # Don't jit a user passed jitted function
+        numba_func = func
+    else:
+
+        @numba.generated_jit(nopython=nopython, nogil=nogil, parallel=parallel)
+        def numba_func(window, *_args):
+            if getattr(np, func.__name__, False) is func or isinstance(
+                func, types.BuiltinFunctionType
+            ):
+
+                def impl(window, *_args):
+                    return func(window, *_args)
+
+                return impl
+            else:
+                jf = numba.jit(func, nopython=nopython)
+
+                def impl(window, *_args):
+                    return jf(window, *_args)
+
+                return impl
+
+    @numba.jit(nopython=nopython, nogil=nogil, parallel=parallel)
+    def roll_apply(
+        values: np.ndarray, begin: np.ndarray, end: np.ndarray, minimum_periods: int,
+    ):
```

Comment:
```suggestion
    ) -> np.ndarray:
```

Might as well annotate return